### DISCUSSION TITLE
Omit Unchecked Box

### DISCUSSION THREAD

**Discussion by hjaber** [react-hook-form/react-hook-form#3820] [Q&A] [answered]
I tried searching but couldn't find a post discussing this, when submitting multiple checkboxes, how can you avoid submitting a checkbox that is not checked as false?
If I check the the second and third box below, the output is below when my goal would only be to return  typeOfCow: Brown Cow and typeOfPen: Blue Pen without submitting typeOfApple: false

{"typeOfApple":false, "typeOfCow":"Brown Cow", "typeOfPen":"Blue Pen"}

    <label>Red Apple</label>
    <input  type="checkbox" name="typeOfApple" value="Red Apple" ref={register}/>
    <label>Brown Cow</label>
    <input type="checkbox" name="typeOfCow" value="Brown Cow" ref={register}>
    <label>Blue Pen</label>
    <input type="checkbox" name="typeOfPen" value="Blue Pen" ref={register}/>

https://codesandbox.io/embed/react-hook-form-watch-v6-forked-zvwlr?file=/src/index.js

**Comment by bluebill1049** [comment, marked as answer]
hook form is opinionated on how checkbox works, for flexible usage, you can consider building a controlled input and wrapped with `Controller`.
